core/domain/role_services.py

```diff
@@ -136,8 +144,9 @@
     feconf.ROLE_ID_ADMIN: [feconf.ROLE_ID_MODERATOR],
     feconf.ROLE_ID_BANNED_USER: [feconf.ROLE_ID_GUEST],
     feconf.ROLE_ID_COLLECTION_EDITOR: [feconf.ROLE_ID_EXPLORATION_EDITOR],
-    feconf.ROLE_ID_EXPLORATION_EDITOR: [feconf.ROLE_ID_GUEST],
+    feconf.ROLE_ID_EXPLORATION_EDITOR: [feconf.ROLE_ID_LEARNER],
```

Comment:
Should learner inherit from GUEST?

Reply:
Sorry didn't get your question properly. It inherits from GUEST to inherit the ACTION_PLAY_ANY_PUBLIC_ACTIVITY.
Does the above hierarchy that I showed in PR description look problematic to you, are you suggesting changes in that?

Reply:
I think learner is inheriting from guest below

Reply:
Ah, I did miss that. :) Thanks.